### PR TITLE
Working, but not sophisticated solution

### PR DESCRIPTION
**Pull request by xBlackyPL**

### CODE REVIEW

File: TreeCrownDetection/MainWindow.xaml.cs [hunk 1]
```diff
@@ -176,19 +168,19 @@ private static unsafe Bitmap ToBitmap(IReadOnlyList<double> rawImage, int width,
             var sliderA = Application.Current.MainWindow.FindName("SliderA") as Slider;
             var sliderB = Application.Current.MainWindow.FindName("SliderB") as Slider;
 
-            int cutOff = 15;
-            int a = System.Convert.ToInt32(sliderA.Value);
-            int b = System.Convert.ToInt32(sliderB.Value);
+            var cutOff = 15;
+            var a = System.Convert.ToInt32(sliderA.Value);
+            var b = System.Convert.ToInt32(sliderB.Value);
```
Comment (szymonjach) [inline TreeCrownDetection/MainWindow.xaml.cs:173]: Rider has done it? I think, that better is to keep explicitly type of variable than use that auto type.

File: TreeCrownDetection/MainWindow.xaml.cs [hunk 2]
```diff
@@ -269,96 +259,90 @@ private async void SaveFileButton_OnClick(object sender, RoutedEventArgs e)
...
+            Console.WriteLine("Saving to file: {0}", dlg.FileName);
+            await SaveToFile(dlg.FileName);
         }
 
-        private async Task SaveToFile(String FileName, bool shapefile)
+        private async Task SaveToFile(string fileName)
         {
-            String FileNameGTiff = null;
+            var fileExtension = fileName.Split('.').ToArray().Last();
+            var shapefile = fileExtension == "shp";
```
Comment (szymonjach) [inline TreeCrownDetection/MainWindow.xaml.cs:276]: imho, better approach would be to pass by argument if it is shapefile than resolving it inside saving function.

File: TreeCrownDetection/MainWindow.xaml.cs (hunk 1 above)
Comment (xBlackyPL) [reply TreeCrownDetection/MainWindow.xaml.cs:173]: It is explicite, you have type on the right side of the assignment.

File: TreeCrownDetection/MainWindow.xaml.cs (hunk 2 above)
Comment (xBlackyPL) [reply TreeCrownDetection/MainWindow.xaml.cs:276]: You have to parse user input somewhere, it does not mater in with layer. In this approach you are not limited to two different save file formats.

**Pull request merged**